components/atom/upload/src/index.js

```diff
@@ -16,70 +18,42 @@ const CLASS_BLOCK_TEXT_SECONDARY = `${CLASS_BLOCK_TEXT}-secondary`
 
 const capitalize = text => text[0].toUpperCase() + text.substr(1)
 
-class AtomUpload extends PureComponent {
-  state = {
-    Dropzone: null
-  }
-
-  loadAsyncReactDropzone() {
-    require.ensure(
-      [],
-      require => {
-        const Dropzone = require('react-dropzone').default
-        this.setState({Dropzone})
-      },
-      'react-dropzone'
-    )
-  }
-
-  componentDidMount() {
-    this.loadAsyncReactDropzone()
-  }
-
-  onDrop = files => {
-    const {onFilesSelection} = this.props
-    onFilesSelection && onFilesSelection(files)
-  }
-
-  renderStatusBlock(status) {
+const AtomUpload = ({
+  status,
+  onFilesSelection = () => {},
+  textExplanation,
+  ...props
+}) => {
+  const renderStatusBlock = status => {
     const classNameIcon = `${BASE_CLASS}-icon${capitalize(status)}`
-    const IconStatus = this.props[`icon${capitalize(status)}`]
-    const textStatus = this.props[`text${capitalize(status)}`]
-    const {textExplanation} = this.props
+    const IconStatus = props[`icon${capitalize(status)}`]
+    const textStatus = props[`text${capitalize(status)}`]
     return (
-      <div
-        className={cx(BASE_CLASS, `${BASE_CLASS}--${status}`)}
-        onClick={this.handleClick}
-      >
+      <div className={cx(BASE_CLASS, `${BASE_CLASS}--${status}`)}>
         <span className={classNameIcon}>{IconStatus}</span>
         <div className={CLASS_BLOCK_TEXT}>
           <h4 className={CLASS_BLOCK_TEXT_MAIN}>{textStatus}</h4>
-          {status === STATUSES.ACTIVE && textExplanation && (
-            <p className={CLASS_BLOCK_TEXT_SECONDARY}>{textExplanation}</p>
-          )}
+          {status === STATUSES.ACTIVE &&
+            textExplanation && (
+              <p className={CLASS_BLOCK_TEXT_SECONDARY}>{textExplanation}</p>
+            )}
         </div>
       </div>
     )
   }
 
-  render() {
-    const {status} = this.props
-    const {Dropzone} = this.state
-    const {onDrop} = this
-
-    if (Object.values(STATUSES).includes(status)) {
-      return (
-        Dropzone && (
-          <Dropzone
-            className={`${BASE_CLASS}-dropzone`}
-            disabled={status !== STATUSES.ACTIVE}
-            onDrop={onDrop}
-          >
-            {this.renderStatusBlock(status)}
-          </Dropzone>
-        )
-      )
-    }
+  if (Object.values(STATUSES).includes(status)) {
+    return (
+      <Suspense fallback={null}>
```

Comment:
👀 I think this could break Server Side Rendering as Suspense is not supported on the server.

Reply:
To fix this, it should be used only behind a state change with a useEffect.

Reply:
Fixed! `Suspense` will never be rendered if `ready` is not true which is set only in `useEffect` that will not be executed in SSR